subsystems/FPIDSubsystem.java

```diff
@@ -0,0 +1,163 @@
+package org.frc5587.lib.subsystems;
+
+import org.frc5587.lib.pid.PID;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.controller.ArmFeedforward;
+import edu.wpi.first.wpilibj.controller.PIDController;
+import edu.wpi.first.wpilibj.SpeedController;
+import edu.wpi.first.wpilibj2.command.PIDSubsystem;
+
+public abstract class FPIDSubsystem extends PIDSubsystem {
+    protected FPIDConstants constants;
+    protected SpeedController[] motors;
+    protected SpeedControllerGroup motorGroup;
+
+    private final DigitalInput limitSwitch;
```

Comment:
possibly delete and instead declare these within the actual subsystems themselves